src/main/java/com/conveyal/r5/transit/TransportNetworkCache.java

```diff
@@ -207,6 +206,10 @@ private TransportNetwork buildNetworkFromBundleZip (String networkId) {
             ZipEntry entry;
             while ((entry = zis.getNextEntry()) != null) {
                 File entryDestination = new File(dataDirectory, entry.getName());
+                if (!entryDestination.toPath().normalize().startsWith(dataDirectory.toPath())) {
```

Comment:
I think I see what this is doing (avoiding directory traversal attacks with `..`) but it may not be obvious to a future reader so probably deserves a comment. The exception message also doesn't explain in what way the entry is bad. If this is not intentional, the message could be expanded to say `An item in this zip file would extract to a location outside the intended destination directory. Refusing for security reasons.` which would also serve to explain the code.

Reply:
I see we've got something related in FileStorageKey#checkForDirectoryTraversal, I wonder if they could be combined somehow. On the other hand, this buildNetworkFromBundleZip is deprecated and only used for an "old-format bundle stored as single ZIP file". Another solution might be to fully deprecate it and just delete the code. I don't know how many of these "old-format bundles" we still have around.